src/core/validations/message.rs

```diff
@@ -424,6 +462,12 @@ pub fn validate_user_data_add_body(body: &UserDataBody) -> Result<(), Validation
         UserDataType::Github => {
             validate_github_username(&body.value)?;
         }
+        UserDataType::UserDataPrimaryAddressEthereum => {
+            validate_user_data_primary_address_ethereum(&body.value)?;
+        }
+        UserDataType::UserDataPrimaryAddressSolana => {
```

Comment:
Please add the check for `ProtocolFeature::PrimaryAddresses` here so the messages will be rejected if they are submitted before the version activates. Please add a test for it as well.

Reply:
done